tests/test_mig_shared_settings.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_shared_settings - unit test of the corresponding mig shared module
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests for the migrid module pointed to in the filename"""
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__))))
+
+from tests.support import TEST_OUTPUT_DIR, MigTestCase, FakeConfiguration, \
+    cleanpath, testmain
+from mig.shared.settings import load_settings, update_settings, \
+    parse_and_save_settings
+
+DUMMY_USER = "dummy-user"
+DUMMY_SETTINGS_DIR = 'dummy_user_settings'
+DUMMY_SETTINGS_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_SETTINGS_DIR)
+DUMMY_SYSTEM_FILES_DIR = 'dummy_system_files'
+DUMMY_SYSTEM_FILES_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_SYSTEM_FILES_DIR)
+DUMMY_TMP_DIR = 'dummy_tmp'
+DUMMY_TMP_FILE = 'settings.mRSL'
+DUMMY_TMP_PATH = os.path.join(TEST_OUTPUT_DIR, DUMMY_TMP_DIR)
+DUMMY_MRSL_PATH = os.path.join(DUMMY_TMP_PATH, DUMMY_TMP_FILE)
+
+DUMMY_USER_INTERFACE = ['V3', 'V42']
+DUMMY_DEFAULT_UI = 'V42'
+DUMMY_CONF = FakeConfiguration(user_settings=DUMMY_SETTINGS_PATH,
+                               mig_system_files=DUMMY_SYSTEM_FILES_PATH,
+                               user_interface=DUMMY_USER_INTERFACE,
+                               new_user_default_ui=DUMMY_DEFAULT_UI)
+
+
+class MigSharedSettings(MigTestCase):
+    """Wrap unit tests for the corresponding module"""
+
+    def test_settings_load_save(self):
+        os.makedirs(os.path.join(DUMMY_SETTINGS_PATH, DUMMY_USER))
+        cleanpath(DUMMY_SETTINGS_DIR, self)
+        os.makedirs(os.path.join(DUMMY_SYSTEM_FILES_PATH, DUMMY_USER))
+        cleanpath(DUMMY_SYSTEM_FILES_DIR, self)
+        os.makedirs(os.path.join(DUMMY_TMP_PATH))
+        cleanpath(DUMMY_TMP_DIR, self)
+
+        settings_mrsl = """
+::EMAIL::
+john@doe.org
+
+::SITE_USER_MENU::
+sharelinks
+people
+peers
+"""
+        tmp_fd = open(DUMMY_MRSL_PATH, 'w')
+        tmp_fd.write(settings_mrsl)
+        tmp_fd.close()
+
+        result = parse_and_save_settings(
+            DUMMY_MRSL_PATH, DUMMY_USER, DUMMY_CONF)
+        assert(result[0] and not result[1])
```

Comment:
We need to be using actual test case assertion methods :) I explained this in extensive review commentary earlier: https://github.com/ucphhpc/migrid-sync/pull/91#discussion_r1705048997

Reply:
Ah, thanks - I completely missed that review in my email pile. I'll take a look and go over it again here :+1: